Water tanks can be wrenched to spread water.

Sunbeamstress opened the pull request:
🆑 Earthcrusher
tweak: Water tanks can be wrenched just like fuel tanks, and will spill water. Slip your enemies!
/🆑 

Wrench the tank and watch the flood!

Flooding amount is determined by how much you set the tank to dispense. I added a multiplier to the add_fluid proc for cosmetic reasons - makes it look more convincingly like a proper flood.

Sunbeamstress committed:
Few changes requested by Zuhayr, plus logic tweaks, message fixes, and comments.

Added an admin log (good call!), tweaked wrenching messages, and caught a hole in processing

Integrates water tanks with the fluid system, letting them be wrenched open.

babydoll commented on code/modules/reagents/reagent_dispenser.dm:
```diff
@@ -78,11 +78,51 @@
+/obj/structure/reagent_dispensers/watertank/attackby(obj/item/weapon/W as obj, mob/user as mob)
+	src.add_fingerprint(user)
+	if (istype(W,/obj/item/weapon/wrench))
+		user.visible_message("\The [user] wrenches \the [src]'s drain [modded ? "shut" : "open"].", \
```
Tap, spigot or faucet is better than drain in all these cases

babydoll commented:
Needs an admin log

MistakeNot4892 commented on code/modules/reagents/reagent_dispenser.dm:
```diff
@@ -78,11 +78,56 @@
+	if (istype(W,/obj/item/weapon/wrench))
+		user.visible_message("\The [user] wrenches \the [src]'s tap [modded ? "shut" : "open"].", \
+			"You wrench [src]'s drain [modded ? "shut" : "open"]")
+		modded = modded ? 0 : 1
```
`modded = !modded`

MistakeNot4892 commented on code/modules/reagents/reagent_dispenser.dm:
```diff
@@ -78,11 +78,56 @@
+		drain_water()
+
+/obj/structure/reagent_dispensers/watertank/Move()
+	if (..() && modded)
```
Move() needs to return, `. = ..()` then check `.`.

MistakeNot4892 commented on code/modules/reagents/reagent_dispenser.dm:
```diff
@@ -78,11 +78,56 @@
+	if(modded)
+		to_chat(user, "<span class='warning'>Someone has wrenched open its tap - it's spilling everywhere!</span>")
+
+/obj/structure/reagent_dispensers/watertank/attackby(obj/item/weapon/W as obj, mob/user as mob)
```
`as obj` and `as mob` are verb forms, pls remove.

MistakeNot4892 commented on code/modules/reagents/reagent_dispenser.dm:
```diff
@@ -78,11 +78,56 @@
+
+/obj/structure/reagent_dispensers/watertank/attackby(obj/item/weapon/W as obj, mob/user as mob)
+	src.add_fingerprint(user)
+	if (istype(W,/obj/item/weapon/wrench))
```
`isWrench(W)`

Sunbeamstress commented:
- Added changes as requested by Zuhayr.
- Put the wrenching messages in the "notice" span class.
- Cut the flooding amount in half. On a whim I tested this in a smaller room (5x5 as opposed to 7x7) and promptly drowned at around 75% drainage. Yikes!
- Added comments. Comments are good!

Sunbeamstress committed:
Added a depth level check to the tank.

PsiOmegaDelta commented on code/modules/reagents/reagent_dispenser.dm:
```diff
@@ -78,11 +78,68 @@
+			"<span class='notice'>You wrench [src]'s drain [modded ? "open" : "shut"].</span>")
+
+		if (modded)
+			log_and_message_admins("opened a water tank at [loc.loc.name], leaking water.")
```
No.
`get_area(loc)`

Sunbeamstress replied:
this is really a sign I need to put on my big boy pants and stop copy/pasting shit

PsiOmegaDelta commented on code/modules/reagents/reagent_dispenser.dm:
```diff
@@ -78,11 +78,68 @@
+	if (..() && modded)
+		drain_water()
+
+/obj/structure/reagent_dispensers/watertank/Move()
```
This will miss situation such as `forceMove()`
Better would be for the tank to listen to its own moved-event while, and only while, it's draining.

Sunbeamstress replied:
Opted to remove the Move() entirely, it wasn't adding any useful behavior (draining water) that wasn't already being covered by Process()

PsiOmegaDelta commented on code/modules/reagents/reagent_dispenser.dm:
```diff
@@ -78,11 +78,68 @@
+		T.add_fluid(W * 100, /datum/reagent/water)
+
+/obj/structure/reagent_dispensers/watertank/examine(mob/user)
+	if(!..(user, 2))
```
Not returning the result of the `..()` call

PsiOmegaDelta commented on code/modules/reagents/reagent_dispenser.dm:
```diff
@@ -78,11 +78,68 @@
+	return ..()
+
+/obj/structure/reagent_dispensers/watertank/Process()
+	if (..() && modded)
```
The only truthy value `/Process()` is likely to return, if any, is `PROCESS_KILL` in which case you should stop what you're doing.

PsiOmegaDelta commented on code/modules/reagents/reagent_dispenser.dm:
```diff
@@ -78,11 +78,68 @@
+
+	// Check for depth first, and pass if the water's too high. A four foot high water tank
+	// cannot jettison water above the level of a grown adult's head!
+	var/turf/T = get_turf(src)
```
The watertank will magically dispense water unto the turf even if it's contained by something.
Is this intended?

PsiOmegaDelta commented on code/modules/reagents/reagent_dispenser.dm:
```diff
@@ -78,11 +78,68 @@
+	// Check for depth first, and pass if the water's too high. A four foot high water tank
+	// cannot jettison water above the level of a grown adult's head!
+	var/turf/T = get_turf(src)
+	if(T.get_fluid_depth() > fill_level)
```
Not handling the case where T might be null

Sunbeamstress committed:
Changes requested by Psi.

Sunbeamstress commented:
This is all good to go, at least I think
